src/coprocessor/dag/expr/compare.rs

```diff
@@ -0,0 +1,294 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::i64;
+use std::cmp::Ordering;
+
+use tipb::expression::ScalarFuncSig;
+
+use coprocessor::codec::{mysql, Datum};
+use super::{StatementContext, Result, FnCall};
+
+impl FnCall {
+    pub fn compare_int(&self,
+                       ctx: &StatementContext,
+                       row: &[Datum],
+                       sig: ScalarFuncSig)
```

Comment:
I think just implement `compare_int`without `ScalarFuncSig` and return Ordering directly like `tidb` did would be more clear for reviewer?